core/common/src/main/java/alluxio/network/protocol/databuffer/NioDirectBufferPool.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.network.protocol.databuffer;
+
+import java.nio.ByteBuffer;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Direct buffer pool.
+ */
+public class NioDirectBufferPool {
+  private static final TreeMap<Integer, LinkedList<ByteBuffer>> BUF_POOL = new TreeMap();
```

Comment:
Do you guys think it's possible just to use some design like DefaultStorageDir? So we just count all the buffer sizes and let netty help us maintain the cache.

Reply:
> `DefaultStorageDir`

Not sure what you mean?

Reply:
Maybe I'm not fully understand the intention of this class, why not just use PooledByteBufAllocator and it has nioBuffer function to return ByteBuffer

Reply:
> We can further improve performance by using `ConcurrentSkipListMap` and `ConcurrentLinkedDeque` and remove the synchronization. This can happen after some benchmarking.

Yes, we could improve it in the next PR, I might need your help, we can sync offline

Reply:
> Do you guys think it's possible just to use some design like DefaultStorageDir? So we just count all the buffer sizes and let netty help us maintain the cache.

Yes, we could add a quota and more metrics for the buffer pool,  we could implement this logic in another PR, we could sync offline for the details.